### DISCUSSION TITLE
Add services for Labels

### DISCUSSION THREAD

**Discussion by Mincka** [frenck/spook#686] [Ideas] [answered]
Hello,
Thank you (and the team) for the newly added support of labels, that's great!
When I read the Beta Release notes, I believed that I would be able to dynamically assign labels to entities.

> Do you have solar panels or home battery? Create a label “⚡️ Heavy Energy Usage” to label devices that consume a lot of energy, you could maybe switch them off when there is not enough battery or solar energy available.

I told myself: "Great idea! I'm going to create an automation to dynamically assign labels to entities based on specific events and conditions". In that case, if there's a high energy consumptions, I can label the entities with “ Heavy Energy Usage”.

Maybe you'll tell me that I could use "Alert" or "Repair" for that but I like the idea to use label instead since it will be more flexible. I could have a card that shows only the entities with a specific label for instance.

Since Spook already support that kind of thing for [Areas](https://spook.boo/areas), I think support for "Labels" could be a great addition. Maybe even for the Core project?

**Comment by frenck** [comment]
Already on it!

**Comment by frenck** [comment]
The following PRs related to this are implemented:

- Detect unknown labels used in automations @frenck (#694)
- Detect unknown labels used in scripts @frenck (#696)
- Add homeassistant.delete_label service @frenck (#712)
- Add homeassistant.create_label service @frenck (#713)
- Add homeassistant.add_label_to_area service @frenck (#715)
- Add homeassistant.add_label_to_device service @frenck (#716)
- Add homeassistant.add_label_to_entity service @frenck (#717)
- Add homeassistant.remove_label_from_area service @frenck (#718)
- Add homeassistant.remove_label_from_device service @frenck (#719)
- Add homeassistant.remove_label_from_entity service @frenck (#720)

Will be part of the v3 release (probably landing upcoming weekend).

../Frenck

**Reply by Mincka** [reply]
Wonderful! Thank you very much. That's a great addition. 👍
Perfect timing with https://github.com/thomasloven/lovelace-auto-entities/releases/tag/v1.13.0 😀

**Comment by MartijnGP** [comment]
Hi Spook! May I suggest more ways to interact with the tag? Adding and removing them dynamically already makes them much more useful, but I would like to be able to filter entities based on their flags as well in my templates. So that I can perform actions on entities depending on the tags they may (or may not) have.

**Reply by Mincka** [reply]
You mean like [this](https://www.home-assistant.io/docs/configuration/templating/#labels)?

**Comment by frenck** [comment, marked as answer]
This is long done, it shipped back in Spook 3.0.0. Sorry for leaving the discussion open this long.

The release notes promise you read is backed by real actions now: create and delete a label, and add or remove a label on an entity, a device or an area. So the dynamic assignment you were after works from an automation.

Docs: https://spook.boo/labels

../Frenck

<p>
  <a href="https://www.linkedin.com/in/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/linkedin.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://youtube.com/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/youtube.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://bsky.app/profile/frenck.social"><img src="https://github.com/frenck/frenck/raw/main/images/bluesky.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://fosstodon.org/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/mastodon.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.instagram.com/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/instagram.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.threads.net/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/threads.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.facebook.com/frenck.dev/"><img src="https://github.com/frenck/frenck/raw/main/images/facebook.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://x.com/frenck"><img src="https://github.com/frenck/frenck/raw/main/images/x.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.tiktok.com/@frenck.nl"><img src="https://github.com/frenck/frenck/raw/main/images/tiktok.svg" width="18" height="18"></a>
</p>

Blogging my personal ramblings at <a href="https://frenck.dev">frenck.dev</a>
